### PR TITLE
Temporarily enforce "Debug build" TRT EP with trt oss parser on Windows

### PR DESCRIPTION
**Pull request by chilo-ms**
This PR handles two changes:

1.  There is an issue when running "Debug build" TRT EP with "Release build" TRT builtin parser on Windows. Enforce use oss parser for Debug build.
Note: args.config in build.py is an array, for example ["Debug", "Release"...]. The code will be much mess if we made the change there. 
2. Update to use latest commit of oss parser.

Please see the https://github.com/microsoft/onnxruntime/issues/16273

### CODE REVIEW

**Review by jywu-msft — APPROVED** [review]
Thanks!

File: cmake/onnxruntime_providers.cmake [hunk 1]
```diff
@@ -694,6 +694,13 @@ if (onnxruntime_USE_TENSORRT)
   endif()
   set(CXX_VERSION_DEFINED TRUE)
 
+  # There is an issue when running "Debug build" TRT EP with "Release build" TRT builtin parser on Windows.
+  # We enforce following workaround for now until the real fix.
+  if (WIN32 AND CMAKE_BUILD_TYPE STREQUAL "Debug")
```
[the remark this answers is not in the document]
Comment (chilo-ms) [reply cmake/onnxruntime_providers.cmake:699]: Do you suggest other better approaches to check whether it's the "Debug build" or not?

[the remark this answers is not in the document]
Comment (snnn) [reply cmake/onnxruntime_providers.cmake:699]: We cannot do it in cmake. Instead, we may use conditional macros like : "$<$<CONFIG:Debug>:SET_DEBUG_MODE>"

[the remark this answers is not in the document]
Comment (chilo-ms) [reply cmake/onnxruntime_providers.cmake:699]: Got it, thanks!

**Review by jywu-msft — APPROVED** [review]

File: cmake/onnxruntime_providers.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (jywu-msft) [reply cmake/onnxruntime_providers.cmake:699]: this is temporary, while the underlying issue can be addressed by nvidia.

**Pull request merged**